tests/testutils/goroutineassistant.go

```diff
@@ -18,23 +18,27 @@ import (
 	"fmt"
 	"sync"
 	"testing"
+	"time"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/steveeJ/gexpect"
 )
 
 type GoroutineAssistant struct {
-	s  chan string
+	s  chan error
 	wg sync.WaitGroup
 	t  *testing.T
 }
 
 func NewGoroutineAssistant(t *testing.T) *GoroutineAssistant {
 	return &GoroutineAssistant{
-		s: make(chan string),
+		s: make(chan error),
 		t: t,
 	}
 }
 
 func (a *GoroutineAssistant) Fatalf(s string, args ...interface{}) {
-	a.s <- fmt.Sprintf(s, args...)
+	a.wg.Done()
+	a.s <- fmt.Errorf(s, args...)
```

Comment:
I have several issues with GoroutineAssistant's Fatalf:

- The `s` channel should have a buffer (say of two elements), so sending a message there is not blocked.
- It should not call Done() on WaitGroup - it is done in every goroutine as a deferred action. The problem was that this deferred action might not be executed, because of being blocked by sending an error to blocking channel. 
- The `return` directives were removed after calls to Fatalf in goroutines, so it should imply that Fatalf does not return (it should call `runtime.Goexit()`).